src/NLog.Web/NLogHttpModule.cs

```diff
@@ -1,56 +1,53 @@
-﻿using System;
+﻿using NLog.Web.Targets.Wrappers;
+using System;
 using System.Web;
 
 namespace NLog.Web
 {
     /// <summary>
-    /// ASP.NET HttpModule that enables NLog to hook BeginRequest and EndRequest events easily.
+    /// ASP.NET IHttpModule that enables AspNetBufferingTargetWrapper proper functioning
     /// </summary>
     public class NLogHttpModule : IHttpModule
     {
-        /// <summary>
-        /// Event to be raised at the end of each HTTP Request.
-        /// </summary>
-        public static event EventHandler EndRequest;
-
-        /// <summary>
-        /// Event to be raised at the beginning of each HTTP Request.
-        /// </summary>
-        public static event EventHandler BeginRequest;
-
         /// <summary>
         /// Initializes the HttpModule.
         /// </summary>
-        /// <param name="application">
+        /// <param name="context">
         /// ASP.NET application.
         /// </param>
-        public void Init(HttpApplication application)
+        public void Init(HttpApplication context)
         {
-            application.BeginRequest += BeginRequestHandler;
-            application.EndRequest += EndRequestHandler;
+            Initialize(context);
```

Comment:
Think you need to change this back, so it becomes:
```
application.BeginRequest += BeginRequestHandler;
```

Reply:
Complete